erpnext/public/js/utils/serial_no_batch_selector.js

```diff
@@ -156,6 +161,11 @@ erpnext.SerialNoBatchSelector = Class.extend({
 			row.warehouse = values.warehouse || warehouse;
 		}
 
+		if(item_code_field) {
+			item_code_field.set_value(this.item_code);
```

Comment:
So yes, `set_value()`, the only way one trigger the remaining fields.
Seems it still has a problem, completely setting only the last row in the group (the first row is already set initially), leaving all the others partially filled:
![set_value_miss](https://user-images.githubusercontent.com/5196925/27571341-a3bd4c96-5b22-11e7-9f63-d3c63aa102b8.gif)
This probably has to do with it being called in bulk for all the items.